src/browser/atom-application.coffee

```diff
@@ -428,12 +428,22 @@ class AtomApplication
 
   saveState: (allowEmpty=false) ->
     return if @quitting
+
+    Config = (require '../config')
+    config = new Config
+      configDirPath: process.env.ATOM_HOME
+      resourcePath: @resourcePath
+    config.setSchema null, {type: 'object', properties: _.clone(require('../config-schema'))}
+    config.load()
```

Comment:
Currently, I believe this will throw an exception when the user's config is invalid, because it will [try to reference](https://github.com/atom/atom/blob/73728f501a9a7cde5b473e38f8e773a1c3efd847/src/config.coffee#L782) `atom.notifications`, which won't be defined in the browser process. If we're going to use the `Config` class here, I think we'll need to remove its dependency on the `NotificationManager`.

That code is also going to change a little bit when https://github.com/atom/atom/pull/8968 lands. Maybe instead of creating the notification inside of `Config`, we should add a method `::onDidFailToLoad(callback)`, and we can use that method in the renderer process to set up a callback that creates the notification.